File: java-funcionario/src/main/java/com/alex/crud/funcionario/FuncionarioService.java
package com.alex.crud.funcionario;

import com.alex.crud.funcionario.model.Funcionario;

import java.util.List;

public interface FuncionarioService {

    Iterable<Funcionario> findAll();

    Funcionario findById(Long id);

    List<Funcionario> findByNome(String nome);

    Funcionario save(Funcionario funcionario);

    Funcionario update(Long id, Funcionario funcionario);

    void delete(Long id);
}

File: angular-funcionario/src/app/services/funcionario.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Funcionario } from '../models/funcionario.model';

const url = 'http://localhost:8080/funcionario';

@Injectable({
  providedIn: 'root'
})
export class FuncionarioService {

  constructor(private http: HttpClient) { }

  findAll(): Observable<Funcionario[]> {
    return this.http.get<Funcionario[]>(url);
  }

  findById(id: any): Observable<Funcionario> {
    return this.http.get(`${url}/${id}`);
  }

  create(data: any): Observable<any> {
    return this.http.post(url, data);
  }

  update(id: number, funcionario: Funcionario): Observable<any> {
    return this.http.put(`${url}/${id}`, funcionario);
  }

  delete(id: any): Observable<any> {
    return this.http.delete(`${url}/${id}`);
  }

  findByNome(nome: string): Observable<Funcionario[]> {
    return this.http.get<Funcionario[]>(`${url}/find-by-nome/${nome}`);
  }
}
